### PR TITLE
pkg/ansible,pkg/helm: import client-go auth plugins

### LINKED ISSUE #1178: panic: No Auth Provider found for name "oidc" when --type helm
## Bug Report

```
operator-sdk new logstash-operator --type helm --helm-chart stable/logstash
operator-sdk up local

INFO[0000] Running the operator locally.                
INFO[0000] Using namespace default.                     
{"level":"info","ts":1551951667.5239906,"logger":"cmd","msg":"Go Version: go1.10.4"}
{"level":"info","ts":1551951667.5240135,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1551951667.5240204,"logger":"cmd","msg":"Version of operator-sdk: v0.5.0+git"}
{"level":"info","ts":1551951667.5240347,"logger":"cmd","msg":"Watching single namespace.","Namespace":"default"}
panic: No Auth Provider found for name "oidc"

goroutine 1 [running]:
github.com/operator-framework/operator-sdk/vendor/k8s.io/client-go/discovery.NewDiscoveryClientForConfigOrDie(0xc42053c000, 0x0)
        src/github.com/operator-framework/operator-sdk/vendor/k8s.io/client-go/discovery/discovery_client.go:454 +0x65
github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDiscoveryRESTMapper(0xc42053c000, 0xc4206dd9e0, 0x0, 0x0, 0x0)
        src/github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/apimachinery.go:35 +0x2f
github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/manager.New(0xc42053c000, 0xc4204d6000, 0x1bfaff8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4207e68b0, ...)
        src/github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go:171 +0x167
github.com/operator-framework/operator-sdk/pkg/helm.Run(0xc4203007e0, 0x0, 0x0)
        src/github.com/operator-framework/operator-sdk/pkg/helm/run.go:72 +0x267
github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up.upLocalHelm(0x1b1206b, 0x4)
        src/github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up/local.go:152 +0x5e
github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up.upLocalFunc(0xc4208aca00, 0x2b69e78, 0x0, 0x0, 0x0, 0x0)
        src/github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up/local.go:97 +0x24d
github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).execute(0xc4208aca00, 0x2b69e78, 0x0, 0x0, 0xc4208aca00, 0x2b69e78)
        src/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:762 +0x468
github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4207cac80, 0xc4208ad400, 0xc4208acc80, 0xc4208ac780)
        src/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4207cac80, 0xc42087ff78, 0xc420046178)
        src/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        src/github.com/operator-framework/operator-sdk/commands/operator-sdk/main.go:24 +0x27
```

**Environment**
* operator-sdk version:
operator-sdk version v0.5.0+git

* Kubernetes version information:

```
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"9e4010d067ee4d799330980e62830b06a07296bd", GitTreeState:"clean", BuildDate:"2018-04-26T04:13:21Z", GoVersion:"go1.10.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2+coreos.0", GitCommit:"b427929b2982726eeb64e985bc1ebb41aaa5e095", GitTreeState:"clean", BuildDate:"2018-01-18T22:56:14Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

Helm operator to handle logstash installations

### LINKED ISSUE #1196: No Auth provider found for name "azure"
## Bug Report

**What did you do?**
I'm trying to run `operator-sdk up local` to test my developed operator.
The kubernetes cluster is running with Azure AKS and uses RBAC authorization with Azure AD.

**What did you expect to see?**
I'm expecting to be able to use the local operator.

**What did you see instead? Under which circumstances?**
I'm getting this exception after running the `operator-sdk up local` command:

```
operator-sdk up local --kubeconfig=/c/Users/N1122966/.kube/config --namespace=default
INFO[0000] Running the operator locally.INFO[0000] Using namespace default.
{"level":"info","ts":1552380484.646539,"logger":"cmd","msg":"Go Version: go1.11.5"}
{"level":"info","ts":1552380484.6473377,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1552380484.647636,"logger":"cmd","msg":"Version of operator-sdk: v0.5.0+git"}
{"level":"info","ts":1552380484.6478565,"logger":"cmd","msg":"Watching single namespace.","Namespace":"default"}
panic: No Auth Provider found for name "azure"

goroutine 1 [running]:
github.com/operator-framework/operator-sdk/vendor/k8s.io/client-go/discovery.NewDiscoveryClientForConfigOrDie(0xc0000e43c0, 0x0)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/vendor/k8s.io/client-go/discovery/discovery_client.go:454 +0x65
github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDiscoveryRESTMapper(0xc0000e43c0, 0xc0003dd050, 0x0, 0x0, 0x0)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/client/apiutil/apimachinery.go:35 +0x2f
github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/manager.New(0xc0000e43c0, 0xc0000d4000, 0x1c25cc8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0006c2310, ...)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go:171 +0x157
github.com/operator-framework/operator-sdk/pkg/helm.Run(0xc000604980, 0x0, 0x0)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/pkg/helm/run.go:72 +0x262
github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up.upLocalHelm(0x1b3c75f, 0x4)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up/local.go:152 +0x5e
github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up.upLocalFunc(0xc000892a00, 0xc000604c00, 0x0, 0x2, 0x0, 0x0)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/up/local.go:97 +0x241
github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).execute(0xc000892a00, 0xc000604bc0, 0x2, 0x2, 0xc000892a00, 0xc000604bc0)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000790c80, 0xc000893400, 0xc000892c80, 0xc000892780)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).Execute(0xc000790c80, 0xc000879f88, 0xc0000ac058)
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /home/n1122966/go/src/github.com/operator-framework/operator-sdk/commands/operator-sdk/main.go:24 +0x27
``` 

**Environment**
* operator-sdk version:
0.5.0

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:37:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.6", GitCommit:"ab91afd7062d4240e95e51ac00a18bd58fddd365", GitTreeState:"clean", BuildDate:"2019-02-26T12:49:28Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster kind: 

Azure AKS Version 12.6

* Are you writing your operator in ansible, helm, or go?

helm

**Possible Solution**
Add the `"plugin/pkg/client/auth/azure",` from `client-go` package in the Gopkg.lock to import the Azure auth provider?

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**
Imports all known Kubernetes `client-go` auth plugins in the Helm and Ansible operators

**Motivation for the change:**
For Go operators, we [plan to add documentation](https://github.com/operator-framework/operator-sdk/issues/710) about how users can add auth plugins to their `cmd/manager/main.go` file.

However, users of the Ansible and Helm operators don't have that option unless the convert their operator to a hybrid operator using `operator-sdk migrate`.

This closes #1178 and closes #1196

### CODE REVIEW

**Comment by AlexNPavel** [comment]
Why don't we add all the auth providers to the go `cmd.go` scaffold as well? We currently add the gcp auth provider by default: https://github.com/operator-framework/operator-sdk/blob/master/pkg/scaffold/cmd.go#L55

Is there a reason to include just the gcp provider by default in the go operators? Or should that be removed and instead be left to the users to handle for go based operators?

/cc @hasbro17

**Comment by joelanford** [comment]
@AlexNPavel Nice, I didn't realize we were doing that. I'll update the PR to include that.

Based on [this comment](https://github.com/operator-framework/operator-sdk/issues/710#issuecomment-436916846), I'll also move the import closer to the top in a separate section, so we don't accidentally break things if we change imports later.

**Comment by hasbro17** [comment]
@AlexNPavel I think originally we thought it's an unnecessary dependency to include auth plugins that you won't use so we were initially intending for users to include them in their Go based operators as needed.

However since more and more users were running into issues on running the operator locally outside GKE we just added the gcp plugin by default.

To be more consistent we can either remove all plugins for Go operators and document clearly what users need to add. Or just add everything for them. I'm leaning towards the latter since it works out of the box.

And for Helm/Ansible operators we'll definitely need them by default since users can't add the imports themselves.

File: pkg/scaffold/cmd.go [hunk 1]
```diff
@@ -43,6 +43,9 @@ import (
 	"os"
 	"runtime"
 
+	// Import all Kubernetes client auth plugins, e.g. Azure, GCP, OIDC, etc.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
```
Comment (hasbro17) [inline pkg/scaffold/cmd.go]: @joelanford Do you think we could confirm this with a manual against GKE?

I've always seen each of the plugins imported separately
```Go
import (
    _ "k8s.io/client-go/plugin/pkg/client/auth/gcp
    _ "k8s.io/client-go/plugin/pkg/client/auth/azure
)
```
Just want to confirm if this method of importing all plugins works as well.

**Comment by joelanford** [comment]
I'm now debating where the imports for the Ansible and Helm operators should go.

In this PR, they're currently in both `pkg/ansible/run.go` and `pkg/helm/run.go`, but I'm now considering if it would be better to have the imports appear only in `main` functions in:
- the hybrid [ansible operator main.go template](https://github.com/operator-framework/operator-sdk/blob/master/pkg/scaffold/ansible/main.go)
- the hybrid [helm operator main.go template](https://github.com/operator-framework/operator-sdk/blob/master/pkg/scaffold/helm/main.go)
- [`commands/operator-sdk/main.go`](https://github.com/operator-framework/operator-sdk/blob/master/commands/operator-sdk/main.go) to cover `operator-sdk up local` and `operator-sdk run`

`go-lint` warns that blank imports should only be in main functions (or have a comment to justify placement elsewhere). Also, putting them in the main functions of hybrid operators would line up more with the go operator and give developers of hybrid operators a hint that auth plugin imports can be tweaked.

Thoughts?

File: pkg/scaffold/cmd.go (hunk 1 above)
Comment (joelanford) [reply pkg/scaffold/cmd.go]: [It should work](https://github.com/kubernetes/client-go/blob/master/plugin/pkg/client/auth/plugins.go), but we should test it to double check and to make sure the imports are in the correct spots.

**Comment by hasbro17** [comment]
@joelanford That seems reasonable to me.
So long as we have `up local` and `run` covered by having the imports defined in `cmd/operator-sdk/main.go`

**Comment by joelanford** [comment]
@hasbro17 @AlexNPavel PTAL

**Pull request merged**